### PR TITLE
Fix/7464 qapublic page source count and preferred source indicator not visible for research resources nor peer reviews

### PR DESCRIPTION
**Pull request by DanielPalafox**

### CODE REVIEW

**Comment by DanielPalafox** [comment]
[[QA][public page] Source count and preferred source indicator not visible for research resources nor peer reviews](https://trello.com/c/grPy3y9q/7464-qapublic-page-source-count-and-preferred-source-indicator-not-visible-for-research-resources-nor-peer-reviews)

**Comment by DanielPalafox** [comment]
[[QA][public page] Item counter not accurate when there are more than 50 research resources](https://trello.com/c/P6bHgrmT/7467-qapublic-page-item-counter-not-accurate-when-there-are-more-than-50-research-resources)

File: src/app/record/components/research-resource-stack/research-resource-stack.component.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+  /**
+   * Show and hide details of the panel
+   */
+  toggleDetails(work: Work) {
+    const putCode = work.putCode.value
+    this.panelDetailsState[putCode].state = !this.panelDetailsState[putCode]
+      .state
+
+    if (this.panelDetailsState[putCode].state) {
```
Comment (leomendoza123) [inline src/app/record/components/research-resource-stack/research-resource-stack.component.ts:120]: Maybe we can remove this (I know I left this on works 🤦‍♂️)

File: src/app/record/components/research-resource-stack/research-resource-stack.component.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        ? this.researchResourceStack.activePutCode.toString() ===
+          researchResource.putCode
+        : false
+    return response
+  }
+
+  /**
+   * Show and hide details of the panel
+   */
+  toggleDetails(work: Work) {
```
Comment (leomendoza123) [inline src/app/record/components/research-resource-stack/research-resource-stack.component.ts:115]: probably not works expected as an attribute here

File: src/app/record/components/research-resource/research-resource.component.ts [hunk 3]
```diff
@@ -1,48 +1,40 @@
...
 
-  userSession: UserSession
-  researchResources: ResearchResourcesEndpoint
-  platform: PlatformInfo
+  @Input() isPublicRecord: string
+  @Input() researchResource: ResearchResource
+  @Input() panelDetailsState: {
+    state: boolean
+  }
+  @Output() toggleDetails = new EventEmitter<Work>()
```
Comment (leomendoza123) [inline src/app/record/components/research-resource/research-resource.component.ts:32]:
```suggestion
  @Output() toggleDetails = new EventEmitter<ResearchResource>()
```

File: src/app/record/components/peer-review-stack/peer-review-stack.component.ts [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+  /**
+   * Show and hide details of the panel
+   */
+  toggleDetails(peerReview: PeerReview) {
+    const putCode = peerReview.putCode.value
+    this.panelDetailsState[putCode].state = !this.panelDetailsState[putCode]
+      .state
+
+    if (this.panelDetailsState[putCode].state) {
```
Comment (leomendoza123) [inline src/app/record/components/peer-review-stack/peer-review-stack.component.ts:132]: can be removed?

File: src/app/record/components/peer-review/peer-review.component.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+})
+export class PeerReviewComponent implements OnInit {
+  @Input() isPublicRecord: string
+  @Input() subPeerReview: PeerReview
+  @Input() panelDetailsState: {
+    state: boolean
+  }
+  @Input() detailsPeerReviews: any
+
+  @Output() toggleDetails = new EventEmitter<PeerReview>()
```
Comment (leomendoza123) [inline src/app/record/components/peer-review/peer-review.component.ts:17]: Seems like this event is never call

File: src/app/record/components/research-resource/research-resource.component.ts (hunk 3 above)
Comment (leomendoza123) [reply src/app/record/components/research-resource/research-resource.component.ts:32]: But this seems like is never call with a next value

File: src/app/record/components/peer-review-stack/peer-review-stack.component.scss [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  word-break: break-word;
+
+  p {
+    margin-top: 0;
+    margin-bottom: 5px;
+  }
+}
+
+.header-sub-peer-review {
+  margin-right: 40px;
```
Comment (leomendoza123) [inline src/app/record/components/peer-review-stack/peer-review-stack.component.scss:32]: remember we need to use `margin-inline-end` so we support RTL

Comment (leomendoza123) [reply src/app/record/components/peer-review-stack/peer-review-stack.component.scss:32]: Same for the other left/right margin and padding

**Review by leomendoza123 — CHANGES_REQUESTED** [review]
Just a few comments

**Comment by DanielPalafox** [comment]
[[QA][public page] Long source-work-id overflows peer review container](https://trello.com/c/W4prHVct/7473-qapublic-page-long-source-work-id-overflows-peer-review-container)

File: src/app/record/components/peer-review-stack/peer-review-stack.component.scss (hunk 6 above)
Comment (DanielPalafox) [reply src/app/record/components/peer-review-stack/peer-review-stack.component.scss:32]: Done!

File: src/app/record/components/research-resource-stack/research-resource-stack.component.ts (hunk 1 above)
Comment (DanielPalafox) [reply src/app/record/components/research-resource-stack/research-resource-stack.component.ts:120]: Yes, you are rigth. I am gonna remove it

**Review by leomendoza123 — APPROVED** [review]

**Pull request merged**